src/core.ts

```diff
@@ -9,6 +9,24 @@ export function registerJoi(customJoi : typeof joi) {
     Joi = customJoi;
 }
 
+export type LabelTransformer = ((propertyKey: string | Symbol, target: object) => string | null | undefined);
+
+let globalLabelTransformer: LabelTransformer | undefined | null = undefined;
```

Comment:
:worried: I'm reluctant to add a global variable, we already have one for the Joi instance and I don't much like that. I don't know off-hand a better way to achieve this, though.

Global variables will limit the ability to use tsdv-joi across multiple dependencies within a containing app. If one dependency desires this label transformer, but another dependency is not expecting it and has quirky logic to inspect errors by the label (for some reason), things could break.

Reply:
Maybe you could pass it as an option to the validator functions/classes?